govcd/system.go

```diff
@@ -479,17 +480,91 @@ func GetExternalNetwork(vcdClient *VCDClient, networkName string) (*ExternalNetw
 
 	if !vcdClient.Client.IsSysAdmin {
 		return Task{}, fmt.Errorf("functionality requires system administrator privileges")
 	}
 
-	err := validateExternalNetwork(externalNetwork)
+	err := validateExternalNetwork(externalNetworkData)
 	if err != nil {
 		return Task{}, err
 	}
 
+	// Type: VimObjectRefType
+	// Namespace: http://www.vmware.com/vcloud/extension/v1.5
+	// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/7a028e78-bd37-4a6a-8298-9c26c7eeb9aa/09142237-dd46-4dee-8326-e07212fb63a8/doc/doc/types/VimObjectRefsType.html
+	// Description: Represents the Managed Object Reference (MoRef) and the type of a vSphere object.
+	// Since: 0.9
+	type vimObjectRefCreate struct {
+		VimServerRef  *types.Reference `xml:"vmext:VimServerRef"`
+		MoRef         string           `xml:"vmext:MoRef"`
+		VimObjectType string           `xml:"vmext:VimObjectType"`
+	}
+
+	// Type: VimObjectRefsType
+	// Namespace: http://www.vmware.com/vcloud/extension/v1.5
+	// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/7a028e78-bd37-4a6a-8298-9c26c7eeb9aa/09142237-dd46-4dee-8326-e07212fb63a8/doc/doc/types/VimObjectRefsType.html
+	// Description: List of VimObjectRef elements.
+	// Since: 0.9
+	type vimObjectRefsCreate struct {
+		VimObjectRef []*vimObjectRefCreate `xml:"vmext:VimObjectRef"`
+	}
+
+	// Type: VMWExternalNetworkType
+	// Namespace: http://www.vmware.com/vcloud/extension/v1.5
+	// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/7a028e78-bd37-4a6a-8298-9c26c7eeb9aa/09142237-dd46-4dee-8326-e07212fb63a8/doc/doc/types/VMWExternalNetworkType.html
+	// Description: External network type.
+	// Since: 1.0
+	type externalNetworkCreate struct {
+		XMLName          xml.Name                    `xml:"vmext:VMWExternalNetwork"`
+		XmlnsVmext       string                      `xml:"xmlns:vmext,attr,omitempty"`
+		XmlnsVcloud      string                      `xml:"xmlns:vcloud,attr,omitempty"`
+		HREF             string                      `xml:"href,attr,omitempty"`
+		Type             string                      `xml:"type,attr,omitempty"`
+		ID               string                      `xml:"id,attr,omitempty"`
+		OperationKey     string                      `xml:"operationKey,attr,omitempty"`
+		Name             string                      `xml:"name,attr"`
+		Link             []*types.Link               `xml:"Link,omitempty"`
+		Description      string                      `xml:"vcloud:Description,omitempty"`
+		Tasks            *types.TasksInProgress      `xml:"Tasks,omitempty"`
+		Configuration    *types.NetworkConfiguration `xml:"vcloud:Configuration,omitempty"`
+		VimPortGroupRef  *vimObjectRefCreate         `xml:"VimPortGroupRef,omitempty"`
+		VimPortGroupRefs *vimObjectRefsCreate        `xml:"vmext:VimPortGroupRefs,omitempty"`
+		VCloudExtension  *types.VCloudExtension      `xml:"VCloudExtension,omitempty"`
+	}
+
+	externalNetwork := &externalNetworkCreate{}
```

Comment:
I request for a big comment explaining what is happening here and, most importantly, answering the question "Why!?"

Reply:
Create needs to use two namespaces and go don't work with them, so namespaces are hardcoded. Read functionality needs seperate struct as API returns different look of xml from create in regards of namespaces(prefix)

Reply:
Thanks for explanation, only my request was to add a comment to code :)

Reply:
Added